sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java

```diff
@@ -514,9 +514,9 @@ public static SharedKeyCredential getSharedKeyCredential(HttpPipeline httpPipeli
                 byte[] cache = new byte[(int) count];
                 int lastIndex = data.read(cache);
                 currentTotalLength[0] += lastIndex;
-                if (currentTotalLength[0] < count) {
```

Comment:
I don't understand what this array of size 1 is (or why it's even an array) and so I don't understand this change.

Reply:
This is because we need to variable to remember the current total length.
However, the lambda will complain the variable must be immutable if having  something like `int sum`. 

This is the best way of recording the current length in lambda, based on the discussion with team and @anuchandy 

The fix here is to compare the chunk we currently process, so it should compare the reading bytes and the expected to have for the chunk, not the currentTotalLength.